code/iaas/service-discovery/api/src/main/java/io/cattle/platform/servicediscovery/api/filter/ServiceAddRemoveLinkServiceValidationFilter.java

```diff
@@ -0,0 +1,59 @@
+package io.cattle.platform.servicediscovery.api.filter;
+
+import io.cattle.platform.core.model.Service;
+import io.cattle.platform.iaas.api.filter.common.AbstractDefaultResourceManagerFilter;
+import io.cattle.platform.servicediscovery.api.constants.ServiceDiscoveryConstants;
+import io.cattle.platform.servicediscovery.api.dao.ServiceConsumeMapDao;
+import io.cattle.platform.util.type.CollectionUtils;
+import io.github.ibuildthecloud.gdapi.request.ApiRequest;
+import io.github.ibuildthecloud.gdapi.request.resource.ResourceManager;
+import io.github.ibuildthecloud.gdapi.validation.ValidationErrorCodes;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.inject.Inject;
+
+public class ServiceAddRemoveLinkServiceValidationFilter extends AbstractDefaultResourceManagerFilter {
+
+    @Inject
+    ServiceConsumeMapDao consumeMapDao;
+
+    private static final Map<String, Boolean> ACTIONS;
+
+    static {
+        ACTIONS = new HashMap<>();
```

Comment:
Quick question: Is the boolean in this map just to drive the logic in validateAction()?
If so, I'd suggest maybe using an enum?

Sort of like the following:
public class foo {

public enum Actions {
 foo, bar
}

public static void main(String[] args) {
  String input = "foo";
  Actions a = Actions.valueOf(input);
  switch(a) {
        case foo:
          System.out.println("execute foo");
          break;
        case bar:
          System.out.println("execute bar");
          break;
  }
}
}

There are also similar examples you can look up online if we have spaces in the action name.